### ISSUE TITLE
Failed to eval optional object attributes

### ISSUE THREAD

**Issue by wata727** [terraform-linters/tflint#1436]
## Introduction

[Optional object type attributes](https://github.com/hashicorp/terraform/pull/26540) are introduced in v0.14 as an experimental language feature. It allows an object variable to have an optional attribute, and if you omit that attribute, its value will be null.

However, TFLint and Terraform have different evaluation results for this omitted value:

```hcl
terraform {
  experiments = [module_variable_optional_attrs]
}

variable "foo" {
  type = object({
    a = optional(string)
    b = optional(string)
  })
}

resource "aws_instance" "foo" {
  instance_type = var.foo.b
}
```

```console
% TF_VAR_foo={a=1} tflint
Failed to check ruleset; Failed to check `aws_instance_previous_type` rule: failed to eval an expression in main.tf:13; Unsupported attribute: This object does not have an attribute named "b".
```

```console
% TF_VAR_foo={a=1} terraform console
╷
│ Warning: Experimental feature "module_variable_optional_attrs" is active
│
│   on main.tf line 2, in terraform:
│    2:   experiments = [module_variable_optional_attrs]
│
│ Experimental features are subject to breaking changes in future minor or patch releases, based on feedback.
│
│ If you have feedback on the design of this feature, please open a GitHub issue to discuss it.
╵

> var.foo.b
tostring(null)
```

## Expected Behavior

Like Terraform, TFLint should return null as a result of `var.foo.b`.

## Actual behavior

It returns the "This object does not have an attribute named "b" error.

### 1. Create `main.tf` and `.tflint.hcl`

```hcl
terraform {
  experiments = [module_variable_optional_attrs]
}

variable "foo" {
  type = object({
    a = optional(string)
    b = optional(string)
  })
}

resource "aws_instance" "foo" {
  instance_type = var.foo.b
}
```

```hcl
plugin "aws" {
  enabled = true
  source = "github.com/terraform-linters/tflint-ruleset-aws"
  version = "0.15.0"
}
```

### 2. Run `tflint --init` and `terraform init`

```console
% tflint --init
% terraform init
```

### 3. Run TFLint and confirm the error

```console
% TF_VAR_foo={a=1} tflint
Failed to check ruleset; Failed to check `aws_instance_previous_type` rule: failed to eval an expression in main.tf:13; Unsupported attribute: This object does not have an attribute named "b".
```

### 4. Run `terraform console` and confirm the behavior

```console
% TF_VAR_foo={a=1} terraform console
╷
│ Warning: Experimental feature "module_variable_optional_attrs" is active
│
│   on main.tf line 2, in terraform:
│    2:   experiments = [module_variable_optional_attrs]
│
│ Experimental features are subject to breaking changes in future minor or patch releases, based on feedback.
│
│ If you have feedback on the design of this feature, please open a GitHub issue to discuss it.
╵

> var.foo.b
tostring(null)
```

## Additional Context

```console
$ tflint -v
TFLint version 0.38.1
+ ruleset.aws (0.15.0)
$ terraform -v
Terraform v1.2.5
on darwin_amd64
+ provider registry.terraform.io/hashicorp/aws v4.22.0
```

**Comment by drewmullen** [comment]
@wata727 wanted to make you aware that Hashi is getting ready to add 1 more feature to the `optional()` featureset: https://github.com/hashicorp/terraform/releases/tag/v1.3.0-alpha20220706

you'll now be able to set a default within your `optional()` set
